src/window.cc

```diff
@@ -656,6 +656,64 @@ void Window::set_menu_actions() {
     SelectionDialog::get()->show();
   });
   
+  menu.add_action("source_find_file", [this]() {
+    auto view = Notebook::get().get_current_view();
+    auto project_path = boost::filesystem::canonical(Directories::get().path);
+  
+    if(view) {
+      auto dialog_iter=view->get_iter_for_dialog();
+      SelectionDialog::create(view, view->get_buffer()->create_mark(dialog_iter), true, true);
+    }
+    else {
+      SelectionDialog::create(true, true);
+    }
+    
+    // std::vector<std::string> excludes;
+    boost::filesystem::path build_default_path, build_debug_path;
+    auto build = Project::Build::create(project_path);
+    if(!project_path.empty()) {
+      if (is_directory(build->get_default_path())) {
+        build_default_path = boost::filesystem::canonical(build->get_default_path());
+      }      
+      if (is_directory(build->get_debug_path())) {
+        build_debug_path = boost::filesystem::canonical(build->get_debug_path());
+      }
+    }
+  
+    // populate with all files in project
+    for (boost::filesystem::recursive_directory_iterator iter(project_path), end; iter != end; iter++) {
+      auto path = boost::filesystem::canonical(iter->path());
```

Comment:
You can remove all uses of `boost::filesystem::canonical` I think as this is taken care of elsewhere in juCi++ in the functions where this is needed.